### PR TITLE
Add TopologicalMesh to use OpenMesh

### PR DESCRIPTION
**Pull request by dlyr**
This PR add the creation of TopologicalMesh, that use OpenMesh, with a converter from/to basic Mesh.
The converter needs improvement to manager vertices attributs, but is yet working for "simple" vertex.
It also add a command line application to show how to convert and use TopologicalMesh and Mesh.

By the way something wet wrong during rebase, 
0d78a46 and d8eb7ee should be one commit with "adding command line application example"
-> FIXED (I rebase again ... git is magic, but I had to force push ...)

### CODE REVIEW

File: Applications/SimpleSubdivideExample/main.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+    else{        
+        Ra::Core::TriangleMesh mesh;
+        Ra::Core::TopologicalMesh topologicalMesh;
+        Ra::Core::OBJFileManager obj;
+        if(a.inputFilename.empty()){
+            mesh  = Ra::Core::MeshUtils::makeBox();
+        }
+        else{
+            obj.load(a.inputFilename, mesh);
```
Comment (hoshiryu) [inline Applications/SimpleSubdivideExample/main.cpp]: Note here that under the current implementation of the OBJFileManager, the mesh is considered triangle-only.
Hence, if the mesh has quadrangular (or other polygon types) faces, the loaded TriangleMesh might endup with holes.

File: src/Core/Mesh/Wrapper/Convert.cpp [hunk 2]
```diff
@@ -132,6 +132,53 @@ void convert( const Dcel& dcel, TriangleMesh& mesh ) {
 }
 
 
+/// Particular conversion for a progressive mesh
+/// since some faces doesn't exist anymore
+void convertPM( const Dcel& dcel, TriangleMesh& mesh )
+{
+
+    // TODO !!!
```
Comment (hoshiryu) [inline src/Core/Mesh/Wrapper/Convert.cpp]: ?

File: src/Tests/CoreTests/Geometry/GeometryTests.hpp [hunk 3]
```diff
@@ -36,7 +36,7 @@ namespace RaTests
 
             // Test line queries.
             RA_UNIT_TEST(  pointToLineSq(a, a, b-a) == 0.f, "distance from A to AB" );
-            RA_UNIT_TEST(  pointToLineSq(b, a, b-a) == 0.f, "distance from B to AB" );
+            RA_UNIT_TEST(  areApproxEqual(pointToLineSq(b, a, b-a), 0.f), "distance from B to AB" );
```
Comment (hoshiryu) [inline src/Tests/CoreTests/Geometry/GeometryTests.hpp]: Why changing this test only and not the first one too?

File: Applications/SimpleSubdivideExample/main.cpp (hunk 1 above)
Comment (dlyr) [reply Applications/SimpleSubdivideExample/main.cpp]: Do not change the app right ? do we need/want to handle quad mesh ? 
I could use Assimp here, but simpler to just call OBJFileManager

File: src/Core/Mesh/Wrapper/Convert.cpp (hunk 2 above)
Comment (dlyr) [reply src/Core/Mesh/Wrapper/Convert.cpp]: (don't know ...)

File: src/Tests/CoreTests/Geometry/GeometryTests.hpp (hunk 3 above)
Comment (dlyr) [reply src/Tests/CoreTests/Geometry/GeometryTests.hpp]: right, I will fix that (i'm only responsible for the rebase/subdivider app, I do not rewrite nor read all the code ;))

Comment (dlyr) [reply src/Tests/CoreTests/Geometry/GeometryTests.hpp]: btw it should be equal or approx equal ?
if approxEqual I should change the segment tests also

File: Applications/SimpleSubdivideExample/main.cpp (hunk 1 above)
Comment (hoshiryu) [reply Applications/SimpleSubdivideExample/main.cpp]: Yeah, just wanted to mention that won't work for quad meshes as is.

**Pull request closed without merging**